src/rdb.c

```diff
@@ -1964,12 +1964,15 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
             unsigned char *data =
                 rdbGenericLoadStringObject(rdb,RDB_LOAD_PLAIN,&encoded_len);
             if (data == NULL || (encoded_len == 0)) {
+                if (encoded_len == 0)
+                    zfree(data);
```

Comment:
nit pick: in theory a more correct check would be `if (data) zfree(data);`
the check on `encoded_len` is just because we know there are two entry conditions to the above if.
but also, since it's valid to zfree a NULL pointer, we can just `zfree(data);` without any condition.